### PR TITLE
Item_2_Sprint_1911: Visualize Image Rotation using OpenCV

### PR DESCRIPTION
**Pull request by AshAswin**

### CODE REVIEW

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
 
-#include <image_transport/image_transport.h>
-#include "std_msgs/String.h"
-#include "sensor_msgs/Image.h"
-#include <sensor_msgs/image_encodings.h>
+#include "../include/image_rotation/image_algorithm.h"
 
 #include <cv_bridge/cv_bridge.h>
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/highgui/highgui.hpp>
```
Comment (srv26) [inline catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: Please check if both the headers of open cv are required .

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp [hunk 2]
```diff
@@ -1,38 +1,41 @@
...
+#include <ros/ros.h>
+#include "sensor_msgs/Image.h"
+#include <sensor_msgs/image_encodings.h>
+#include "std_msgs/String.h"
+
 static const std::string OPENCV_WINDOW = "Image Window";
 
 void ImageRotationCallback(const sensor_msgs::ImageConstPtr& msg)
 {
-    utilities::helloworld();
```
Comment (srv26) [inline catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: Whats the use ?

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp [hunk 3]
```diff
@@ -1,38 +1,41 @@
...
 
 #include <cv_bridge/cv_bridge.h>
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/highgui/highgui.hpp>
 
+#include <image_transport/image_transport.h>
+#include <ros/ros.h>
+#include "sensor_msgs/Image.h"
+#include <sensor_msgs/image_encodings.h>
+#include "std_msgs/String.h"
```
Comment (srv26) [inline catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: whats the use?

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp [hunk 4]
```diff
@@ -1,38 +1,41 @@
...
-#include "std_msgs/String.h"
-#include "sensor_msgs/Image.h"
-#include <sensor_msgs/image_encodings.h>
+#include "../include/image_rotation/image_algorithm.h"
 
 #include <cv_bridge/cv_bridge.h>
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/highgui/highgui.hpp>
 
+#include <image_transport/image_transport.h>
```
Comment (melisamariam) [inline catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp:9]: Check if image transport can be used here.

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp [hunk 5]
```diff
@@ -1,38 +1,41 @@
...
+    cv::imshow(OPENCV_WINDOW, cv_ptr->image);
     cv::waitKey(3);
 }
 
 int  main(int argc, char* argv[])
 { 
     ros::init(argc, argv, "ImageSubscriberNode");
     ros::NodeHandle image_subscriber_handle; 
-    ros::Subscriber image_subscriber = image_subscriber_handle.subscribe("/sensors/camera/image_color",1000,ImageRotationCallback);
+    ros::Subscriber image_subscriber = image_subscriber_handle.
```
Comment (melisamariam) [inline catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: check if image transport should be used

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp (hunk 4 above)
Comment (melisamariam) [reply catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp:9]: Check the following link http://wiki.ros.org/image_transport

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp (hunk 5 above)
Comment (melisamariam) [reply catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: Check the following link http://wiki.ros.org/image_transport

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp (hunk 1 above)
Comment (AshAswin) [reply catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: Both the headers are not required and therefore are removed

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp (hunk 3 above)
Comment (AshAswin) [reply catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: String header file inclusion removed

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp (hunk 4 above)
Comment (AshAswin) [reply catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp:9]: ros namespace for the subscriber replaced with the image_transport namespace

File: catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp (hunk 5 above)
Comment (AshAswin) [reply catkin_ws/src/image_rotation/src/image_rotation_subscriber.cpp]: ros namespace for the subscriber replaced with the image_transport namespace

**Review by melisamariam — APPROVED** [review]

**Pull request merged**